src/main/java/net/sf/jabref/gui/keyboard/KeyBindingsDialogViewModel.java

```diff
@@ -0,0 +1,196 @@
+                                super.updateItem(item, empty);
+                                KeyBindingViewModel viewModel = getTreeTableRow().getItem();
+                                if ((viewModel != null) && !viewModel.isCategory()) {
+                                    Text graphic = new Text(IconTheme.JabRefIcon.CLEANUP_ENTRIES.getCode());
+                                    graphic.getStyleClass().add("icon");
+                                    setGraphic(graphic);
+                                    setOnMouseClicked(evt -> {
+                                        resetKeyBinding(getTreeTableRow().getItem());
+                                    });
+
+                                }
+                                keyBindingsTable.refresh();
+                            }
+                        };
+                    }
+                });
+    }
+
+    private void resetKeyBinding(KeyBindingViewModel keyBindingViewModel) {
+        if (keyBindingViewModel.getKeyBinding() != null) {
+            String key = keyBindingViewModel.getKeyBinding().getKey();
+            keyBindingRepository.resetToDefault(key);
+            keyBindingViewModel.setBinding(keyBindingRepository.get(key));
+        }
+
+    }
+
+    private void registerKeyEvents() {
+        keyBindingsTable.setOnKeyPressed(evt -> grabKey(evt));
+    }
+
+    private Stage getDialogStage() {
+        return (Stage) closeButton.getScene().getWindow();
+    }
+
+    @FXML
+    private void closeDialog() {
+        Stage stage = getDialogStage();
+        stage.close();
+    }
+
+    @FXML
+    private void saveKeyBindingsAndCloseDialog() {
+        keyBindingPreferences.setNewKeyBindings(keyBindingRepository.getKeyBindings());
+
+        String title = Localization.lang("Key bindings changed");
+        String content = Localization.lang("Your new key bindings have been stored.") + '\n'
+                + Localization.lang("You must restart JabRef for the new key bindings to work properly.");
+        FXDialogs.showInformationDialog(title, content);
+        closeDialog();
+    }
+
+    @FXML
+    private void setDefaultBindings() {
+        String title = Localization.lang("Resetting all key bindings");
+        String content = Localization.lang("All key bindings will be reset to their defaults.");
+        ButtonType resetButtonType = new ButtonType("Reset", ButtonData.OK_DONE);
+        FXDialogs.showCustomButtonDialogAndWait(AlertType.INFORMATION, title, content, resetButtonType,
+                ButtonType.CANCEL).ifPresent(response -> {
+                    if (response == resetButtonType) {
+                        keyBindingRepository.resetToDefault();
+                        populateTable();
+                    }
+                });
+    }
+
+    private void grabKey(KeyEvent evt) {
+        // first check if a valid entry is selected
+        if (keyBindingsTable.getSelectionModel().getSelectedIndex() < 0) {
```

Comment:
Just create a new field `selectedKeyBinding` and bind it to keyBindingsTable.getSelectionModel().selectedItemProperty().